models/purchase.py

```diff
@@ -0,0 +1,33 @@
+from odoo import models, fields
+from odoo.exceptions import UserError
+from datetime import timedelta
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    active = fields.Boolean(
+        'Is Active?',
+        default=True,
+        readonly=True,
+        states={'done': [('readonly', False)], 'cancel': [
+            ('readonly', False)]},
```

Comment:
No needs to define `readonly` and `states` here. In stead of doing this, you should perform a check on whether users can archive a Purchase Order or not